### PR TITLE
CHG: [droid] refactor wake lock; follow our screensaver

### PR DESCRIPTION
**Pull request by koying**
This refactors the wake lock to release it when our screensaver goes active.

This allows:
- Control on the wake lock via our screensaver settings
- Devices to go to sleep when we are inactive (e.g. Amazon FTV and tablets)

/cc @Montellese @davilla @t-nelson

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
I got feedback. There's indeed no way to wake from daydream using non controllers. 
So we could integrate this with a warning on Android about daydream.

**Comment by koying** [comment]
I've tweaked the logic.
Now, daydream/sleep will only kick in if our sreensaver is "dim" or "black". If our screensaver is disabled OR set to a 3rd party one, the wakelock will stay, as previously.

That should mitigate the issues.

**Comment by Montellese** [comment]
Did you miss my still unresolved comment from a few months ago? Or is that change correct?

File: xbmc/android/activity/XBMCApp.cpp [hunk 1]
```diff
@@ -230,10 +228,6 @@ void CXBMCApp::onDestroyWindow()
     XBMC_DestroyDisplay();
     XBMC_Pause(true);
   }
-
-  if (m_wakeLock)
-    m_wakeLock->release();
```
Comment (koying) [inline xbmc/android/activity/XBMCApp.cpp]: I've moved it to "OnPause", which is the mirror of "OnResume" where we grab the lock.
I prefer to use those as they directly map to the droid activity lifecycle, so it's clear when those happen.

**Comment by koying** [comment]
jenkins build this please

**Pull request merged**